js/piwik.js

```diff
@@ -5759,9 +5762,30 @@ if (typeof window.Matomo !== 'object') {
              * @param bool
              */
             this.setSecureCookie = function (enable) {
+                if(location.protocol !== 'https:') {
+                    logConsoleError("Error in setSecureCookie: You cannot use `Secure` on http.");
```

Comment:
@fealXX should we maybe return and not set `configCookieIsSecure =true` when it is not on HTTPS?

Also I think we need to change the if to 

```js
if(enable && location.protocol !== 'https:') {
    logConsoleError('...')
   return;
}
```

No need to check for https when `enable=false`

Reply:
Done!